api/src/pcapi/core/offers/api.py

```diff
@@ -277,6 +278,30 @@ def update_offer(
     return offer
 
 
+def update_educational_offer(
+    offer,
+    offer_as_dict,
+) -> Offer:
+    validation.check_validation_status(offer)
+    for key, value in offer_as_dict.items():
+        if key == "extraData":
+            extra_data = copy.deepcopy(offer.extraData)
+
+            for extra_data_key, extra_data_value in value.items():
+                extra_data[extra_data_key] = extra_data_value
+
+            setattr(offer, key, extra_data)
```

Comment:
Ici tu peux même mettre `offer.extra_data = extra_data` du coup :)

Reply:
Je vais essayer de proposer un truc un peu plus "pythonistas" :) avec les list comprehensions version dictionnaire

Reply:
`{k, offer_as_dict.get(k, offer.getattr(k)) for k in set(offer_as_dict) | set(offer.dict())}`
Un truc comme ça? Tu en penses quoi @cgaunet ?
Bon après ça devient moins lisible 😅 mais un peu plus pythonist :)
(Après à voir si on peut directement faire `ofer.dict()` sur le modèle, je n'ai pas testé)